pkg/controllers/node/termination/controller.go

```diff
@@ -138,6 +152,56 @@ func (c *Controller) removeFinalizer(ctx context.Context, n *v1.Node) error {
 	return nil
 }
 
+func (c *Controller) terminationGracePeriodExpirationTime(ctx context.Context, node *v1.Node) (*time.Time, error) {
+
+	nodeClaim := &v1beta1.NodeClaim{}
+
+	if len(node.OwnerReferences) == 0 {
+		logging.FromContext(ctx).Errorf("node has no owner, could not find NodeClaim for Node: %v", node.Name)
+		return nil, nil
+	}
+
+	nodeClaimName := types.NamespacedName{}
+	for _, o := range node.OwnerReferences {
+		// verify that we're finding Karpenter's NodeClaim of any version permutation
+		if strings.HasPrefix(o.APIVersion, "karpenter.sh") && o.Kind == "NodeClaim" {
+			nodeClaimName = types.NamespacedName{
+				Name: o.Name,
+			}
+		}
+	}
+
+	if err := c.kubeClient.Get(ctx, nodeClaimName, nodeClaim); err != nil {
+		return nil, err
```

Comment:
If we don't fine the NodeClaim here and get a NotFound error (for whatever reason), does that just mean that we should return `nil` and ignore the terminationGracePeriod rather than continually erroring when we try to get the result of this function?

Reply:
`golangci-lint` hates the idea of this.
```
pkg/controllers/node/termination/controller.go:183:3: error is not nil (line 181) but it returns nil (nilerr)
		return nil, nil
		^
```